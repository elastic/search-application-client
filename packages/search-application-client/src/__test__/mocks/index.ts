export const ALL_RESULTS = {
  took: 2,
  timed_out: false,
  _shards: {
    total: 2,
    successful: 2,
    skipped: 0,
    failed: 0,
  },
  hits: {
    total: {
      value: 9,
      relation: 'eq',
    },
    max_score: 0,
    hits: [
      {
        _index: 'search-movies',
        _id: 'tt0118971',
        _score: 0,
        _source: {
          Title: "The Devil's Advocate",
          Rated: 'R',
          Plot: 'A hotshot lawyer gets more than he bargained for when he learns his new boss is Lucifer himself.',
          Awards: '1 win & 5 nominations.',
          Poster:
            'http://ia.media-imdb.com/images/M/MV5BMTcyMzI3NDM4Nl5BMl5BanBnXkFtZTcwMTUxOTYzNA@@._V1_SX300.jpg',
          imdbID: 'tt0118971',
          Type: 'movie',
          Response: 'True',
          PosterS3:
            'https://s3-eu-west-1.amazonaws.com/imdbimages/images/MV5BMTcyMzI3NDM4Nl5BMl5BanBnXkFtZTcwMTUxOTYzNA@@._V1_SX300.jpg',
          id: 'tt0118971',
          language: ['English', 'German', 'Italian', 'Mandarin', 'Spanish'],
          genres: ['Drama', 'Mystery', 'Thriller'],
          actors: [
            'Keanu Reeves',
            'Al Pacino',
            'Charlize Theron',
            'Jeffrey Jones',
          ],
          country: ['USA', 'Germany'],
          directors: ['Taylor Hackford'],
          writers: ['Andrew Neiderman', 'Jonathan Lemkin ', 'Tony Gilroy '],
          year: 1997,
          metascore: 60,
          runtime: 144,
          imdbRating: 7.5,
          imdbVotes: 198065,
          released: '1997-10-16T23:00:00.000Z',
        },
      },
      {
        _index: 'search-movies',
        _id: 'tt0111257',
        _score: 0,
        _source: {
          Title: 'Speed',
          Rated: 'R',
          Plot: 'A young cop must prevent a bomb exploding aboard a city bus by keeping its speed above 50 mph.',
          Awards: 'Won 2 Oscars. Another 12 wins & 15 nominations.',
          Poster:
            'http://ia.media-imdb.com/images/M/MV5BNzczNDQyMTc2MF5BMl5BanBnXkFtZTgwNjI2NzQxMTE@._V1_SX300.jpg',
          imdbID: 'tt0111257',
          Type: 'movie',
          Response: 'True',
          PosterS3:
            'https://s3-eu-west-1.amazonaws.com/imdbimages/images/MV5BNzczNDQyMTc2MF5BMl5BanBnXkFtZTgwNjI2NzQxMTE@._V1_SX300.jpg',
          id: 'tt0111257',
          language: ['English'],
          genres: ['Action', 'Crime', 'Thriller'],
          actors: [
            'Keanu Reeves',
            'Dennis Hopper',
            'Sandra Bullock',
            'Joe Morton',
          ],
          country: ['USA'],
          directors: ['Jan de Bont'],
          writers: ['Graham Yost'],
          year: 1994,
          metascore: 78,
          runtime: 116,
          imdbRating: 7.2,
          imdbVotes: 202346,
          released: '1994-06-09T23:00:00.000Z',
        },
      },
      {
        _index: 'search-movies',
        _id: 'tt0103874',
        _score: 0,
        _source: {
          Title: 'Dracula',
          Rated: 'R',
          Plot: "The vampire comes to England to seduce a visitor's fiancée and inflict havoc in the foreign land.",
          Awards: 'Won 3 Oscars. Another 9 wins & 13 nominations.',
          Poster:
            'http://ia.media-imdb.com/images/M/MV5BMTI5MTU0NjAyMl5BMl5BanBnXkFtZTcwNTM3MjM1MQ@@._V1_SX300.jpg',
          imdbID: 'tt0103874',
          Type: 'movie',
          Response: 'True',
          PosterS3:
            'https://s3-eu-west-1.amazonaws.com/imdbimages/images/MV5BMTI5MTU0NjAyMl5BMl5BanBnXkFtZTcwNTM3MjM1MQ@@._V1_SX300.jpg',
          id: 'tt0103874',
          language: ['English', 'Romanian', 'Greek', 'Bulgarian', 'Latin'],
          genres: ['Drama', 'Fantasy', 'Horror'],
          actors: [
            'Gary Oldman',
            'Winona Ryder',
            'Anthony Hopkins',
            'Keanu Reeves',
          ],
          country: ['USA'],
          directors: ['Francis Ford Coppola'],
          writers: ['Bram Stoker', 'James V. Hart '],
          year: 1992,
          metascore: 57,
          runtime: 128,
          imdbRating: 7.5,
          imdbVotes: 110033,
          released: '1992-11-13T00:00:00.000Z',
        },
      },
      {
        _index: 'search-movies',
        _id: 'tt0191397',
        _score: 0,
        _source: {
          Title: 'The Replacements',
          Rated: 'PG-13',
          Plot: 'During a pro football strike, the owners hire substitute players.',
          Awards: null,
          Poster:
            'http://ia.media-imdb.com/images/M/MV5BMTcwODQ4NDEzMl5BMl5BanBnXkFtZTcwMTIwMTgyMQ@@._V1_SX300.jpg',
          imdbID: 'tt0191397',
          Type: 'movie',
          Response: 'True',
          PosterS3:
            'https://s3-eu-west-1.amazonaws.com/imdbimages/images/MV5BMTcwODQ4NDEzMl5BMl5BanBnXkFtZTcwMTIwMTgyMQ@@._V1_SX300.jpg',
          id: 'tt0191397',
          language: ['English', 'American Sign Language'],
          genres: ['Comedy', 'Sport'],
          actors: [
            'Keanu Reeves',
            'Gene Hackman',
            'Brooke Langton',
            'Orlando Jones',
          ],
          country: ['USA'],
          directors: ['Howard Deutch'],
          writers: ['Vince McKewin'],
          year: 2000,
          metascore: 30,
          runtime: 118,
          imdbRating: 6.5,
          imdbVotes: 38977,
          released: '2000-08-10T23:00:00.000Z',
        },
      },
      {
        _index: 'search-movies',
        _id: 'tt0114887',
        _score: 0,
        _source: {
          Title: 'A Walk in the Clouds',
          Rated: 'PG-13',
          Plot: "After returning from the war, Paul and a young woman meet on a bus as she's headed home from college to help with the grape harvest and face her Old World domineering dad. The woman has not...",
          Awards: 'Won 1 Golden Globe. Another 3 nominations.',
          Poster:
            'http://ia.media-imdb.com/images/M/MV5BNjg3OTQzOTM5MF5BMl5BanBnXkFtZTYwNTY5NzY5._V1_SX300.jpg',
          imdbID: 'tt0114887',
          Type: 'movie',
          Response: 'True',
          PosterS3:
            'https://s3-eu-west-1.amazonaws.com/imdbimages/images/MV5BNjg3OTQzOTM5MF5BMl5BanBnXkFtZTYwNTY5NzY5._V1_SX300.jpg',
          id: 'tt0114887',
          language: ['English'],
          genres: ['Drama', 'Romance'],
          actors: [
            'Keanu Reeves',
            'Aitana Sánchez-Gijón',
            'Anthony Quinn',
            'Giancarlo Giannini',
          ],
          country: ['USA', 'Mexico'],
          directors: ['Alfonso Arau'],
          writers: [
            'Robert Mark Kamen ',
            'Mark Miller ',
            'Harvey Weitzman ',
            'Piero Tellini',
            'Cesare Zavattini',
            'Vittorio de Benedetti',
          ],
          year: 1995,
          metascore: null,
          runtime: 102,
          imdbRating: 6.6,
          imdbVotes: 19542,
          released: '1995-08-10T23:00:00.000Z',
        },
      },
      {
        _index: 'search-movies',
        _id: 'tt0133093',
        _score: 0,
        _source: {
          Title: 'The Matrix',
          Rated: 'R',
          Plot: 'A computer hacker learns from mysterious rebels about the true nature of his reality and his role in the war against its controllers.',
          Awards: 'Won 4 Oscars. Another 29 wins & 36 nominations.',
          Poster:
            'http://ia.media-imdb.com/images/M/MV5BMTkxNDYxOTA4M15BMl5BanBnXkFtZTgwNTk0NzQxMTE@._V1_SX300.jpg',
          imdbID: 'tt0133093',
          Type: 'movie',
          Response: 'True',
          PosterS3:
            'https://s3-eu-west-1.amazonaws.com/imdbimages/images/MV5BMTkxNDYxOTA4M15BMl5BanBnXkFtZTgwNTk0NzQxMTE@._V1_SX300.jpg',
          id: 'tt0133093',
          language: ['English'],
          genres: ['Action', 'Adventure', 'Sci-Fi'],
          actors: [
            'Keanu Reeves',
            'Laurence Fishburne',
            'Carrie-Anne Moss',
            'Hugo Weaving',
          ],
          country: ['USA', 'Australia'],
          directors: ['Andy Wachowski', 'Lana Wachowski'],
          writers: ['Andy Wachowski', 'Lana Wachowski'],
          year: 1999,
          metascore: 73,
          runtime: 136,
          imdbRating: 8.7,
          imdbVotes: 824103,
          released: '1999-03-30T23:00:00.000Z',
        },
      },
      {
        _index: 'search-movies',
        _id: 'tt0091860',
        _score: 0,
        _source: {
          Title: "River's Edge",
          Rated: 'R',
          Plot: "A high school slacker kills his girlfriend and shows off her dead body to his friends. However, the friends' reaction is almost as ambiguous and perplexing as the crime itself.",
          Awards: '3 wins & 5 nominations.',
          Poster:
            'http://ia.media-imdb.com/images/M/MV5BMTcxNzQ4Mzc1Ml5BMl5BanBnXkFtZTcwMzMxNDMzMQ@@._V1_SX300.jpg',
          imdbID: 'tt0091860',
          Type: 'movie',
          Response: 'True',
          PosterS3:
            'https://s3-eu-west-1.amazonaws.com/imdbimages/images/MV5BMTcxNzQ4Mzc1Ml5BMl5BanBnXkFtZTcwMzMxNDMzMQ@@._V1_SX300.jpg',
          id: 'tt0091860',
          language: ['English'],
          genres: ['Crime', 'Drama'],
          actors: [
            'Crispin Glover',
            'Keanu Reeves',
            'Ione Skye',
            'Daniel Roebuck',
          ],
          country: ['USA'],
          directors: ['Tim Hunter'],
          writers: ['Neal Jimenez'],
          year: 1986,
          metascore: null,
          runtime: 99,
          imdbRating: 7.1,
          imdbVotes: 9722,
          released: '1987-05-07T23:00:00.000Z',
        },
      },
      {
        _index: 'search-movies',
        _id: 'tt0101452',
        _score: 0,
        _source: {
          Title: "Bill & Ted's Bogus Journey",
          Rated: 'PG',
          Plot: 'A tyrant from the future creates evil android doubles of Bill and Ted and sends them back to eliminate the originals.',
          Awards: '1 win & 1 nomination.',
          Poster:
            'http://ia.media-imdb.com/images/M/MV5BMjAwNjQyODA2Ml5BMl5BanBnXkFtZTcwNDM4MzcyNA@@._V1_SX300.jpg',
          imdbID: 'tt0101452',
          Type: 'movie',
          Response: 'True',
          PosterS3:
            'https://s3-eu-west-1.amazonaws.com/imdbimages/images/MV5BMjAwNjQyODA2Ml5BMl5BanBnXkFtZTcwNDM4MzcyNA@@._V1_SX300.jpg',
          id: 'tt0101452',
          language: ['English', 'Italian', 'Russian'],
          genres: ['Adventure', 'Comedy', 'Fantasy'],
          actors: [
            'Keanu Reeves',
            'Alex Winter',
            'William Sadler',
            'Joss Ackland',
          ],
          country: ['USA'],
          directors: ['Peter Hewitt'],
          writers: [
            'Chris Matheson',
            'Ed Solomon',
            'Chris Matheson',
            'Ed Solomon',
          ],
          year: 1991,
          metascore: 60,
          runtime: 93,
          imdbRating: 6.1,
          imdbVotes: 40137,
          released: '1991-07-18T23:00:00.000Z',
        },
      },
      {
        _index: 'search-movies',
        _id: 'tt0970416',
        _score: 0,
        _source: {
          Title: 'The Day the Earth Stood Still',
          Rated: 'PG-13',
          Plot: 'A remake of the 1951 classic sci-fi film about an alien visitor and his giant robot counterpart who visit Earth.',
          Awards: '3 wins & 5 nominations.',
          Poster:
            'http://ia.media-imdb.com/images/M/MV5BNzMyNjE0MTcxNF5BMl5BanBnXkFtZTcwODcyOTMwMg@@._V1_SX300.jpg',
          imdbID: 'tt0970416',
          Type: 'movie',
          Response: 'True',
          PosterS3:
            'https://s3-eu-west-1.amazonaws.com/imdbimages/images/MV5BNzMyNjE0MTcxNF5BMl5BanBnXkFtZTcwODcyOTMwMg@@._V1_SX300.jpg',
          id: 'tt0970416',
          language: ['English', 'Mandarin'],
          genres: ['Drama', 'Sci-Fi', 'Thriller'],
          actors: [
            'Keanu Reeves',
            'Jennifer Connelly',
            'Kathy Bates',
            'Jaden Smith',
          ],
          country: ['USA', 'Canada'],
          directors: ['Scott Derrickson'],
          writers: ['David Scarpa ', 'Edmund H. North'],
          year: 2008,
          metascore: 40,
          runtime: 104,
          imdbRating: 5.5,
          imdbVotes: 115183,
          released: '2008-12-12T00:00:00.000Z',
        },
      },
    ],
  },
  aggregations: {
    my_custom_agg: {
      count: 9,
      min: 1986,
      max: 2008,
      avg: 1995.7777777777778,
      sum: 17962,
    },
    year_facet: {
      count: 9,
      min: 1986,
      max: 2008,
      avg: 1995.7777777777778,
      sum: 17962,
    },
    directors_facet: {
      doc_count_error_upper_bound: 0,
      sum_other_doc_count: 0,
      buckets: [
        {
          key: 'Alfonso Arau',
          doc_count: 1,
        },
        {
          key: 'Andy Wachowski',
          doc_count: 1,
        },
        {
          key: 'Francis Ford Coppola',
          doc_count: 1,
        },
        {
          key: 'Howard Deutch',
          doc_count: 1,
        },
        {
          key: 'Jan de Bont',
          doc_count: 1,
        },
        {
          key: 'Lana Wachowski',
          doc_count: 1,
        },
        {
          key: 'Peter Hewitt',
          doc_count: 1,
        },
        {
          key: 'Scott Derrickson',
          doc_count: 1,
        },
        {
          key: 'Taylor Hackford',
          doc_count: 1,
        },
        {
          key: 'Tim Hunter',
          doc_count: 1,
        },
      ],
    },
  },
}

export const AGG_ACTORS_RESPONSE = {
  took: 3,
  timed_out: false,
  _shards: {
    total: 2,
    successful: 2,
    skipped: 0,
    failed: 0,
  },
  hits: {
    total: {
      value: 4162,
      relation: 'eq',
    },
    max_score: null,
    hits: [],
  },
  aggregations: {
    actors_facet: {
      doc_count_error_upper_bound: 24,
      sum_other_doc_count: 15818,
      buckets: [
        {
          key: 'Naveen Andrews',
          doc_count: 73,
        },
        {
          key: 'Jennifer Carpenter',
          doc_count: 56,
        },
        {
          key: 'Michael C. Hall',
          doc_count: 56,
        },
        {
          key: 'Emilie de Ravin',
          doc_count: 53,
        },
        {
          key: 'Jared Padalecki',
          doc_count: 49,
        },
        {
          key: 'Jensen Ackles',
          doc_count: 49,
        },
        {
          key: 'Hugh Laurie',
          doc_count: 41,
        },
        {
          key: 'Julie Benz',
          doc_count: 40,
        },
        {
          key: 'Henry Ian Cusick',
          doc_count: 39,
        },
        {
          key: 'Matthew Fox',
          doc_count: 34,
        },
      ],
    },
  },
}
