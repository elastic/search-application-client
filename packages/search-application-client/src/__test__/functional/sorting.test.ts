import nock from 'nock'
import Client from '../../index'
import 'cross-fetch/polyfill'
import { ALL_RESULTS } from '../mocks'

describe('Sorting', () => {
  it('should send sorting parameter', async () => {
    const client = Client('test', 'http://localhost:6000', 'test', {
      facets: {},
    })

    nock('http://localhost:6000')
      .post(
        '/_application/search_application/test/_search',
        (requestBody: any) => {
          expect(requestBody.params._es_sort_fields).toEqual([
            { year: 'asc' },
            '_score',
          ])
          return true
        }
      )
      .times(1)
      .reply(200, ALL_RESULTS)

    const results = await client()
      .setSort([{ year: 'asc' }, '_score'])
      .query('test')
      .search()

    expect(results['response']).toMatchSnapshot('Response with results')
  })
})
