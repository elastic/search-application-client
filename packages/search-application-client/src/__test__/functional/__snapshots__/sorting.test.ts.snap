// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sorting should send sorting parameter: Response with results 1`] = `undefined`;
